### ISSUE TITLE
a5337a6 breaks tests

### ISSUE THREAD

**Issue by terotil** [npm/node-tar#55]
Test run for 5da515d passes but a5337a6 does not (nor does v2.0.0 for that matter)

```
tero@fasaani:~/src/node-tar$ git checkout 5da515d
Note: checking out '5da515d'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b new_branch_name

HEAD is now at 5da515d... Emit finish from Extract when finished.
tero@fasaani:~/src/node-tar$ npm run test

> tar@1.0.3 test /home/tero/src/node-tar
> tap test/*.js

fixtures/
fixtures/200.tar
fixtures/200ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
fixtures/200L.hex
fixtures/200longlink.tar
fixtures/200longname.tar
fixtures/a.hex
fixtures/a.tar
fixtures/a.txt
fixtures/b.hex
fixtures/b.tar
fixtures/b.txt
fixtures/c.hex
fixtures/c.tar
fixtures/c.txt
fixtures/cc.txt
fixtures/dir/
fixtures/dir.tar
fixtures/foo.hex
fixtures/foo.js
fixtures/foo.tar
fixtures/hardlink-1
fixtures/hardlink-2
fixtures/omega.hex
fixtures/omega.tar
fixtures/omega.txt
fixtures/omegapax.tar
fixtures/packtest/
fixtures/r/
fixtures/Ω.txt
fixtures/r/e/
fixtures/r/e/a/
fixtures/r/e/a/l/
fixtures/r/e/a/l/l/
fixtures/r/e/a/l/l/y/
fixtures/r/e/a/l/l/y/-/
fixtures/r/e/a/l/l/y/-/d/
fixtures/r/e/a/l/l/y/-/d/e/
fixtures/r/e/a/l/l/y/-/d/e/e/
fixtures/r/e/a/l/l/y/-/d/e/e/p/
fixtures/r/e/a/l/l/y/-/d/e/e/p/-/
fixtures/r/e/a/l/l/y/-/d/e/e/p/-/f/
fixtures/r/e/a/l/l/y/-/d/e/e/p/-/f/o/
fixtures/r/e/a/l/l/y/-/d/e/e/p/-/f/o/l/
fixtures/r/e/a/l/l/y/-/d/e/e/p/-/f/o/l/d/
fixtures/r/e/a/l/l/y/-/d/e/e/p/-/f/o/l/d/e/
fixtures/r/e/a/l/l/y/-/d/e/e/p/-/f/o/l/d/e/r/
fixtures/r/e/a/l/l/y/-/d/e/e/p/-/f/o/l/d/e/r/-/
fixtures/r/e/a/l/l/y/-/d/e/e/p/-/f/o/l/d/e/r/-/p/
fixtures/r/e/a/l/l/y/-/d/e/e/p/-/f/o/l/d/e/r/-/p/a/
fixtures/r/e/a/l/l/y/-/d/e/e/p/-/f/o/l/d/e/r/-/p/a/t/
tar: Ignoring unknown extended header keyword 'SCHILY.dev'
tar: Ignoring unknown extended header keyword 'SCHILY.ino'
tar: Ignoring unknown extended header keyword 'SCHILY.nlink'
tar: Ignoring unknown extended header keyword 'SCHILY.dev'
tar: Ignoring unknown extended header keyword 'SCHILY.ino'
tar: Ignoring unknown extended header keyword 'SCHILY.nlink'
fixtures/r/e/a/l/l/y/-/d/e/e/p/-/f/o/l/d/e/r/-/p/a/t/h/
fixtures/r/e/a/l/l/y/-/d/e/e/p/-/f/o/l/d/e/r/-/p/a/t/h/cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
fixtures/packtest/omega.txt
fixtures/packtest/star.4.html
tar: Ignoring unknown extended header keyword 'SCHILY.dev'
tar: Ignoring unknown extended header keyword 'SCHILY.ino'
tar: Ignoring unknown extended header keyword 'SCHILY.nlink'
fixtures/packtest/Ω.txt
fixtures/dir/sub/
ok test/00-setup-fixtures.js ............................ 4/4
slow chmod /home/tero/src/node-tar/test/tmp/extract-test/dir
slow utimes /home/tero/src/node-tar/test/tmp/extract-test/dir
slow chmod /home/tero/src/node-tar/test/tmp/extract-test/dir/sub
slow utimes /home/tero/src/node-tar/test/tmp/extract-test/dir/sub
ok test/extract-move.js ................................. 5/5
ok test/extract.js .................................... 45/45
ok test/header.js ..................................... 22/22
ok test/pack-no-proprietary.js ...................... 201/201
ok test/pack.js ..................................... 204/204
ok test/parse.js ...................................... 43/43
ok test/zz-cleanup.js ................................... 3/3
total ............................................... 527/527

ok
tero@fasaani:~/src/node-tar$ git checkout master
Previous HEAD position was 5da515d... Emit finish from Extract when finished.
Switched to branch 'master'
Your branch is up-to-date with 'origin/master'.
tero@fasaani:~/src/node-tar$ npm run test

> tar@2.0.0 test /home/tero/src/node-tar
> tap test/*.js

fixtures/
fixtures/200.tar
fixtures/200ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
fixtures/200L.hex
fixtures/200longlink.tar
fixtures/200longname.tar
fixtures/a.hex
fixtures/a.tar
fixtures/a.txt
fixtures/b.hex
fixtures/b.tar
fixtures/b.txt
fixtures/c.hex
fixtures/c.tar
fixtures/c.txt
fixtures/cc.txt
fixtures/dir/
fixtures/dir.tar
fixtures/foo.hex
fixtures/foo.js
fixtures/foo.tar
fixtures/hardlink-1
fixtures/hardlink-2
fixtures/omega.hex
fixtures/omega.tar
fixtures/omega.txt
fixtures/omegapax.tar
fixtures/packtest/
fixtures/r/
fixtures/Ω.txt
fixtures/r/e/
fixtures/r/e/a/
fixtures/r/e/a/l/
fixtures/r/e/a/l/l/
fixtures/r/e/a/l/l/y/
fixtures/r/e/a/l/l/y/-/
fixtures/r/e/a/l/l/y/-/d/
fixtures/r/e/a/l/l/y/-/d/e/
fixtures/r/e/a/l/l/y/-/d/e/e/
fixtures/r/e/a/l/l/y/-/d/e/e/p/
fixtures/r/e/a/l/l/y/-/d/e/e/p/-/
fixtures/r/e/a/l/l/y/-/d/e/e/p/-/f/
fixtures/r/e/a/l/l/y/-/d/e/e/p/-/f/o/
tar: Ignoring unknown extended header keyword 'SCHILY.dev'
tar: Ignoring unknown extended header keyword 'SCHILY.ino'
tar: Ignoring unknown extended header keyword 'SCHILY.nlink'
tar: Ignoring unknown extended header keyword 'SCHILY.dev'
tar: Ignoring unknown extended header keyword 'SCHILY.ino'
tar: Ignoring unknown extended header keyword 'SCHILY.nlink'
fixtures/r/e/a/l/l/y/-/d/e/e/p/-/f/o/l/
fixtures/r/e/a/l/l/y/-/d/e/e/p/-/f/o/l/d/
fixtures/r/e/a/l/l/y/-/d/e/e/p/-/f/o/l/d/e/
fixtures/r/e/a/l/l/y/-/d/e/e/p/-/f/o/l/d/e/r/
fixtures/r/e/a/l/l/y/-/d/e/e/p/-/f/o/l/d/e/r/-/
fixtures/r/e/a/l/l/y/-/d/e/e/p/-/f/o/l/d/e/r/-/p/
fixtures/r/e/a/l/l/y/-/d/e/e/p/-/f/o/l/d/e/r/-/p/a/
fixtures/r/e/a/l/l/y/-/d/e/e/p/-/f/o/l/d/e/r/-/p/a/t/
fixtures/r/e/a/l/l/y/-/d/e/e/p/-/f/o/l/d/e/r/-/p/a/t/h/
fixtures/r/e/a/l/l/y/-/d/e/e/p/-/f/o/l/d/e/r/-/p/a/t/h/cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
fixtures/packtest/omega.txt
fixtures/packtest/star.4.html
tar: Ignoring unknown extended header keyword 'SCHILY.dev'
tar: Ignoring unknown extended header keyword 'SCHILY.ino'
tar: Ignoring unknown extended header keyword 'SCHILY.nlink'
fixtures/packtest/Ω.txt
fixtures/dir/sub/
ok test/00-setup-fixtures.js ............................ 4/4
not ok test/dir-normalization.js ........................ 7/9
    Command: "/usr/bin/nodejs dir-normalization.js"
    TAP version 13
    ok 1 cleaned!
    ok 2 tar entry 1 fixtures/
    ok 3 tar entry 2 fixtures/the-chumbler
    ok 4 should see 2 entries
    ok 5 unpacked file 1
    ok 6 unpacked file 2 /fixtures
    not ok 7 unpacked file 3 /fixtures/the-chumbler
      ---
        file:   /home/tero/src/node-tar/test/dir-normalization.js
        line:   131
        column: 9
        stack:  
          - |
            getCaller (/home/tero/src/node-tar/node_modules/tap/lib/tap-assert.js:439:17)
          - |
            assert (/home/tero/src/node-tar/node_modules/tap/lib/tap-assert.js:21:16)
          - |
            equivalent (/home/tero/src/node-tar/node_modules/tap/lib/tap-assert.js:183:12)
          - |
            Function.similar (/home/tero/src/node-tar/node_modules/tap/lib/tap-assert.js:284:10)
          - |
            Test._testAssert (/home/tero/src/node-tar/node_modules/tap/lib/tap-test.js:87:16)
          - |
            DirReader.foundEntry (/home/tero/src/node-tar/test/dir-normalization.js:131:9)
          - |
            DirReader.emit (events.js:95:17)
          - |
            DirReader.emitEntry (/home/tero/src/node-tar/node_modules/fstream/lib/dir-reader.js:249:8)
          - |
            LinkReader.EMITCHILD (/home/tero/src/node-tar/node_modules/fstream/lib/dir-reader.js:158:12)
          - |
            LinkReader.emit (events.js:95:17)
        found:  
          path:     /fixtures/the-chumbler
          mode:     120777
          type:     SymbolicLink
          depth:    2
          size:     76
          linkpath: /home/tero/src/node-tar/test/tmp/dir-normalization-test/a/b/c/d/the-chumbler
          nlink:    1
        wanted: 
          path:     /fixtures/the-chumbler
          mode:     120755
          type:     SymbolicLink
          depth:    2
          size:     95
          linkpath: /home/tero/src/node-tar/test/tmp/dir-normalization-test/a/b/c/d/the-chumbler
          nlink:    1
        diff:   |
          {
            "path" : "/fixtures/the-chumbler",
            "mode" : "120777", // != "120755"
            "type" : "SymbolicLink",
            "depth" : 2,
            "size" : 76, // != 95
            "linkpath" : "/home/tero/src/node-tar/test/tmp/dir-normalization-test/a/b/c/d/the-chumbler",
            "nlink" : 1
          }
      ...
    ok 8 should have 3 items
    not ok 9 test/dir-normalization.js
      ---
        exit:    1
        command: "/usr/bin/nodejs dir-normalization.js"
      ...
    
    1..9
    # tests 9
    # pass  7
    # fail  2
    
slow chmod /home/tero/src/node-tar/test/tmp/extract-test/dir
slow utimes /home/tero/src/node-tar/test/tmp/extract-test/dir
slow chmod /home/tero/src/node-tar/test/tmp/extract-test/dir/sub
slow utimes /home/tero/src/node-tar/test/tmp/extract-test/dir/sub
ok test/extract-move.js ................................. 5/5
ok test/extract.js .................................... 45/45
ok test/header.js ..................................... 22/22
ok test/pack-no-proprietary.js ...................... 201/201
ok test/pack.js ..................................... 204/204
ok test/parse.js ...................................... 43/43
ok test/zz-cleanup.js ................................... 3/3
total ............................................... 534/536

not ok
```

**Comment by othiym23** [comment]
What version of Node.js / io.js are you running? On what OS?

**Comment by terotil** [comment]
Debian GNU/Linux 8.0

```
tero@fasaani:~$ uname -a
Linux fasaani 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt7-1 (2015-03-01) x86_64 GNU/Linux
tero@fasaani:~$ node --version
v0.10.29
```

**Comment by othiym23** [comment]
This is fixed by 96355141e005fa192b4fd4c3134ec3bb824dfca8. Thanks for the report!
